File: modules/agent/agents/sisyphus.go
package agents

import (
	"github.com/woyin/OrangeCoding/modules/agent"
	"github.com/woyin/OrangeCoding/modules/ai"
	"github.com/woyin/OrangeCoding/modules/core"
	"github.com/woyin/OrangeCoding/modules/tools"
)

// NewSisyphus creates the primary general-purpose coding agent.
// RoleCoder, all tools, general system prompt.
func NewSisyphus(provider ai.AiProvider, registry *tools.ToolRegistry, workDir string) *BaseAgent {
	sid := core.NewSessionId()
	agentCtx := agent.NewAgentContext(sid, workDir)
	agentCtx.SetSystemPrompt("You are Sisyphus, a general-purpose coding agent. You write, debug, review, and refactor code. You are thorough, methodical, and never give up on a task.")

	executor := agent.NewToolExecutor(registry)
	toolDefs := agent.BuildToolDefinitions(registry)
	loop := agent.NewAgentLoop(core.NewAgentId(), provider, executor, agentCtx, agent.DefaultLoopConfig(), toolDefs)

	return NewBaseAgent(core.RoleCoder, loop)
}

File: packages/agent/src/agents/sisyphus.ts
/**
 * Sisyphus - primary general-purpose coding agent.
 * Ported from modules/agent/agents/sisyphus.go.
 */

import type { AiProvider } from "@orangecoding/ai";
import { AgentId, SessionId, AgentRole } from "@orangecoding/core";
import { ToolRegistry } from "@orangecoding/tools";
import { AgentContext } from "../context.js";
import { ToolExecutor } from "../executor.js";
import { buildToolDefinitions } from "../tool-defs.js";
import { AgentLoop, defaultLoopConfig } from "../loop.js";
import { BaseAgent } from "./base.js";

export function newSisyphus(provider: AiProvider, registry: ToolRegistry, workDir: string): BaseAgent {
  const sid = SessionId.create();
  const agentCtx = new AgentContext(sid, workDir);
  agentCtx.setSystemPrompt("You are Sisyphus, a general-purpose coding agent. You write, debug, review, and refactor code. You are thorough, methodical, and never give up on a task.");

  const executor = new ToolExecutor(registry);
  const toolDefs = buildToolDefinitions(registry);
  const loop = new AgentLoop(AgentId.create(), provider, executor, agentCtx, defaultLoopConfig(), toolDefs);

  return new BaseAgent(AgentRole.Coder, loop);
}
